### PR TITLE
Add 'object' to built-in props

### LINKED ISSUE #637: prop.object() prop type
```js
import { Component, prop } from 'skatejs';

export default class extends Component {
  static props = {
    myObj: prop.object()
  }
}
```

Defaults options:

- `default` should be a function that returns an object
- `deserialize` should use `JSON.parse()`
- `serialize` should use `JSON.stringify()`
- `coerce` shouldn't do anything since it's hard to infer the meaning of the input

### PR DESCRIPTION
**Pull request by zzarcon**
Closes #637

### CODE REVIEW

File: src/api/prop.js [hunk 1]
```diff
@@ -42,3 +42,9 @@ export const string = create({
   deserialize: toNullOrString,
   serialize: toNullOrString
 });
+
+export const object = create({
+  default: {},
```
Comment (treshugart) [inline src/api/prop.js]: Should be a function that returns an object (`() => ({})`). This currently would return the same object for all props using this.

File: src/api/prop.js [hunk 2]
```diff
@@ -42,3 +42,9 @@ export const string = create({
...
+  deserialize: JSON.parse,
```
Comment (treshugart) [inline src/api/prop.js]: See `prop.array()` for what should probably be done here. They can share this function, too.

File: test/unit/api/properties.js [hunk 3]
```diff
@@ -184,6 +184,36 @@ describe('api/prop', () => {
     });
   });
 
+  describe('object', () => {
+    let elem;
+
+    beforeEach((done) => {
+      elem = create(prop.object());
+      afterMutations(done, 1);
```
Comment (treshugart) [inline test/unit/api/properties.js]: I know you got this call from somewhere else, but it doesn't take the `1` argument. Just need `done`.

Comment (zzarcon) [inline test/unit/api/properties.js]: My fault, removed the other occurrences as well.

**Comment by zzarcon** [comment]
@treshugart 👀

**Comment by treshugart** [comment]
Can you use `git cz` please and do a `feat` commit? Or squash and merge with `feat` please?

**Comment by zzarcon** [comment]
@treshugart done :crossed_fingers:

**Pull request merged**